src/Appwrite/Platform/Workers/Functions.php

```diff
@@ -282,6 +283,14 @@ private function execute(
 
         $runtime = $runtimes[$function->getAttribute('runtime')];
 
+        $jwtExpiry = $function->getAttribute('timeout', 900);
+        $jwtObj = new JWT(System::getEnv('_APP_OPENSSL_KEY_V1'), 'HS256', $jwtExpiry, 10);
+        $apiKey = $jwtObj->encode([
+            'projectId' => $project->getId(),
+            'scopes' => $function->getAttribute('scopes', [])
+        ]);
```

Comment:
Shouldn't we save this API key to the db as well? How do we validate this key?

Reply:
If you are worried about using the token when it's not expired but resource is deleted, this wont occur, because you would need to delete function or a project. We don't have a way to delete a key currently.